### PR TITLE
Add: MCP3008

### PR DESCRIPTION
**Pull request by brentru**
Adding support for the [MCP3008](https://www.adafruit.com/product/856) analog-to-digital converter. 

- Establishing a base API (mcp3xxx)
- Adding an internal MCP3008 class which inherits mcp3xxx
- add value/volts reading for both single/differential reads

### CODE REVIEW

**Comment by brentru** [comment]
@ladyada we're ready merge and PR!

File: .pylintrc [hunk 1]
```diff
@@ -52,8 +52,7 @@ confidence=
...
-disable=print-statement,parameter-unpacking,unpacking-in-except,old-raise-syntax,backtick,long-suffix,old-ne-operator,old-octal-literal,import-star-module-level,raw-checker-failed,bad-inline-option,locally-disabled,locally-enabled,file-ignored,suppressed-message,useless-suppression,deprecated-pragma,apply-builtin,basestring-builtin,buffer-builtin,cmp-builtin,coerce-builtin,execfile-builtin,file-builtin,long-builtin,raw_input-builtin,reduce-builtin,standarderror-builtin,unicode-builtin,xrange-builtin,coerce-method,delslice-method,getslice-method,setslice-method,no-absolute-import,old-division,dict-iter-method,dict-view-method,next-method-called,metaclass-assignment,indexing-exception,raising-string,reload-builtin,oct-method,hex-method,nonzero-method,cmp-method,input-builtin,round-builtin,intern-builtin,unichr-builtin,map-builtin-not-iterating,zip-builtin-not-iterating,range-builtin-not-iterating,filter-builtin-not-iterating,using-cmp-argument,eq-without-hash,div-method,idiv-method,rdiv-method,exception-message-attribute,invalid-str-codec,sys-max-int,bad-python3-import,deprecated-string-function,deprecated-str-translate-call,import-error
```
Comment (tannewt) [inline .pylintrc]: Don't change this.

File: adafruit_mcp3xxx/adafruit_mcp3xxx.py [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+
+    @property
+    def value(self):
+        """Returns the value from a differential read across two pins as an integer.
+        """
+        diff_pin = MCP3008_DIFF_PINS.get((self._pin_1, self._pin_2), "Difference pin not found.")
+        return self._adc._read_pin_differential(diff_pin)
+
+    @property
+    def volts(self):
```
Comment (tannewt) [inline adafruit_mcp3xxx/adafruit_mcp3xxx.py]: `voltage` not `volts` please. https://circuitpython.readthedocs.io/en/3.x/docs/design_guide.html#sensor-properties-and-units

**Comment by brentru** [comment]
This PR has been replaced by a different implementation. PR: https://github.com/adafruit/Adafruit_CircuitPython_MCP3xxx/pull/4

**Pull request closed without merging**